huaweicloud/resource_huaweicloud_scm_certificate_v3.go

```diff
@@ -0,0 +1,408 @@
+package huaweicloud
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/huaweicloud/golangsdk"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/huaweicloud/golangsdk/openstack/scm/v3/certificates"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+const (
+	MAX_ERROR_MESSAGE_LEN = 200
+	ELLIPSIS_STRING       = "..."
+
+	TARGET_SERVICE_CDN         = "CDN"
+	TARGET_SERVICE_WAF         = "WAF"
+	TARGET_SERVICE_ENHANCE_ELB = "Enhance_ELB"
+)
+
+func resourceScmCertificateV3() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceScmCertificateV3Create,
+		Update: resourceScmCertificateV3Update,
+		Read:   resourceScmCertificateV3Read,
+		Delete: resourceScmCertificateV3Delete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Read:   schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"certificate_chain": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: utils.SuppressNewLineDiffs,
+			},
+			"private_key": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: utils.SuppressNewLineDiffs,
+			},
+			"certificate": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: utils.SuppressNewLineDiffs,
+			},
+			"push_certificate": {
```

Comment:
suggest name as:
target {
  service = "xxx"
  project = ["aaa", "bbb"]
}

Reply:
got it